### PR TITLE
Add Section for Docs regarding SCS Developers

### PR DESCRIPTION
**Pull request by maxwolfs**
This PR adds a second section for documentation regarding the development of SCS, especially targeting SCS Developers. This makes it easier to distinguish between the target groups / roles of Operators and Developers. The main `docs` path in the files system, is for the operator docs. The developer docs are 'dev-docs'. This is reflected in the urls and routing. 

Currently there are only two IAM documents, where there is still the question if they aren't part of the standards drafts? see https://github.com/SovereignCloudStack/docs/pull/72 @reqa  

The [SCS Zuul User Docs](https://github.com/SovereignCloudStack/docs/pull/54 ) should also reside in the path of `dev-docs`.

Additionally the main nav is now directly addressing the roles / personas.

This will make it easier to set up the `index` page of https://docs.scs.community where we – SIG Documentation – want to establish starting points for the different site sections.

Thanks @fkr for the nice session!

### CODE REVIEW

File: dev-docs/operations/iam/identity-federation-in-scs.md
Comment (fkr) [inline dev-docs/operations/iam/identity-federation-in-scs.md:1]: Will this lead to 404 errors?

**Review by fkr — APPROVED** [review]
LGTM - please see question regarding the filename change.

File: dev-docs/operations/iam/identity-federation-in-scs.md
Comment (maxwolfs) [reply dev-docs/operations/iam/identity-federation-in-scs.md:1]: It would yes, but those pages have not been listed anywhere on the docs page yet.

Comment (fkr) [reply dev-docs/operations/iam/identity-federation-in-scs.md:1]: I was actually more worried about links from outside (eg. blogposts, other docs). I grepped through our stuff and didn't find any references. imho good to go. 👍🏻

**Review by master-caster — COMMENTED** [review]
I'm probably not perfectly qualified to do a full featured review on this topic.
LGTM as far as i can tell. 

Adding Zuul quickstart guide to dev-docs works for me.
Is there a task left for me regarding the quick start guide?

**Comment by maxwolfs** [comment]
> I'm probably not perfectly qualified to do a full featured review on this topic. LGTM as far as i can tell.
> 
> Adding Zuul quickstart guide to dev-docs works for me. Is there a task left for me regarding the quick start guide?

Yes, once this is merged the zuul quickstart guide has to be moved to dev-docs then.

**Comment by maxwolfs** [comment]
> > I'm probably not perfectly qualified to do a full featured review on this topic. LGTM as far as i can tell.
> > Adding Zuul quickstart guide to dev-docs works for me. Is there a task left for me regarding the quick start guide?
> 
> Yes, once this is merged the zuul quickstart guide has to be moved to dev-docs then.

I did it now within the last commit.

**Pull request merged**